### PR TITLE
add gateway.istio.io/managed: istio.io-gateway-controller to managed gateway

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

waypoint has "gateway.istio.io/managed": "istio.io-mesh-controller", like that sometimes an operator need to simply look up a label to distinguish if this is istio managed

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
Can you add a PR description explaining why we want this?

**Comment by hzxuzhonghu** [comment]
done

**Comment by howardjohn** [comment]
Ah, I recall now we actually set these, but here: https://github.com/istio/istio/blob/e9cd972401361ec6aae978a86a18270c6517fcf4/pilot/pkg/config/kube/gateway/deploymentcontroller.go#L561

**Comment by hzxuzhonghu** [comment]
Let me have a look

File: pilot/pkg/config/kube/gateway/testdata/deployment/cluster-ip.yaml [hunk 1]
```diff
@@ -50,6 +50,7 @@ spec:
         prometheus.io/port: "15020"
         prometheus.io/scrape: "true"
       labels:
+        gateway.istio.io/managed: istio.io-gateway-controller
```
Comment (hzxuzhonghu) [inline pilot/pkg/config/kube/gateway/testdata/deployment/cluster-ip.yaml:53]: Ping @howardjohn take a look again on the ut

**Review by howardjohn — APPROVED** [review]

**Pull request merged**